src/docs/extensions.md

```diff
@@ -2,16 +2,27 @@
+In the following, we will give you a quick overview of the available extension mechanisms and provide more details in the sections below.
 
-An extension is consumed by adding a dependency to the
-npm-package in the `package.json` of the application.
-Extensions can be installed/uninstalled at
-runtime, which will trigger a recompilation and restart.
+- **VS Code extensions**: Simple to write, installable at runtime, compatible with VS Code, limited to the VS Code extension API, some use cases not possible due to API restrictions => Used for adding features to existing tools. Note that you can also use existing VS Code extensions within Theia, too.
+- **Theia extensions**: Install at compile time, full access to internals of Theia via dependency injection, almost no limitations in terms of accessible API => used to build custom products and for features not covered by the VS Code extension API. Note that the Theia project (including the core) is fully built using Theia extensions in a modular way.
+- **Theia plugins**: Like VS Code extensions, additional access to some Theia specific APIs and the Frontend (frontend plugins), Theia specific parts are not compatible with VS Code.
+The following diagram shows the high level architecture for all three options. VS Code extensions and Theia plugins run in a dedicated process, can be installed at runtime and work against a defined API. Theia extensions are added during compile time and become a first class citizen within your Theia application. They can access the full API of Theia.
 
-Through a DI module, the extension can provide bindings from types to concrete
-implementations, i.e. provide services and contributions.
+<img src="/extensiontypes.png" alt="Theia Logo Blue" style="max-width: 525px">
+
+If you would like more guidance on which mechanism to use, please also refer to [this detailed comparions between VS Code extensions and Theia extensions](https://eclipsesource.com/blogs/2021/03/24/vs-code-extensions-vs-theia-extensions/).
+
+##Theia extensions
+A Theia extension is a module that resides inside a Theia application and communicates with other modules (Theia extensions) via dependency injection. The Theia project itself is composed of Theia extensions too. To create a Theia application, you can select a number of Theia extensions provided by the Theia project, add your own custom Theia extensions and then compile and run the result. Your custom Theia extension will have the same level of access as the core ones. This modularity allows you to extend, adapt or remove almost anything in Theia according to your requirements. Also specific use cases, such as complex views are easier to develop with Theia extensions compared to VS Code extensions.
```

Comment:
> A Theia extension is a module that resides inside a Theia application and communicates with other modules (Theia extensions) via dependency injection.

Note sure I follow what you mean by "communicates via dependency injection".

Reply:
I just removed this, it is not important here